builtin/logical/transit/path_encrypt.go

```diff
@@ -127,6 +127,82 @@ to the min_encryption_version configured on the key.`,
 	}
 }
 
+// decodeBatchRequestItems is a fast path alternative to mapstructure.Decode to decode []BatchRequestItem.
+func decodeBatchRequestItems(src interface{}, dst *[]BatchRequestItem) error {
+	if src == nil || dst == nil {
+		return nil
+	}
+
+	items, ok := src.([]interface{})
+	if !ok {
+		return fmt.Errorf("source data must be an array or slice, got %T", src)
+	}
+
+	// Early return should happen before allocating the array if the batch is empty.
+	// However to comply with mapstructure output we must allocate an empty array.
+	sitems := len(items)
+	*dst = make([]BatchRequestItem, sitems)
+	if sitems == 0 {
+		return nil
+	}
+
+	var errs mapstructure.Error
```

Comment:
Instead of `mapstructure.Error`, why not use `multierror.Error`, since we aren't using `mapstructure` any more?

Reply:
I agree.
However, I've made those changes in the optic of unrolling the logic of `mapstructure` and provide the **exact** same output to be a drop-in-place replacement.
This behaviour is tested in `TestTransit_decodeBatchRequestItems`, using `multierror.Error` instead will fail the test, as expected. 
 
If this isn't wanted and it's preferred to totally removing the `mapstructure, the struct would also need to be stripped from `mapstructure` tags and another set of tests need to be written. 

I don't have strong opinions on the approach here, so far I preferred the path of less surprise and fewer modifications. 

This also applies to https://github.com/hashicorp/vault/pull/8775#discussion_r437663979 and https://github.com/hashicorp/vault/pull/8775#discussion_r437662933.

Reply:
I guess I'm ok with that justification in the context of this PR, we can always rework those bits later.  Could I ask you to add a comment to `errs` explaining this (or linking to this discussion) please?

Reply:
We don't promise that error messages remain consistent so I think `multierror.Error` is fine to move to if desired.